include/cereal/archives/json.hpp

```diff
@@ -235,10 +235,12 @@ namespace cereal
       void saveValue(int i)                 { itsWriter.Int(i);                                                          }
       //! Saves a uint to the current node
       void saveValue(unsigned u)            { itsWriter.Uint(u);                                                         }
+#if LONG_MAX == 0x7FFFFFFFL
       //! Saves a long to the current node
       void saveValue(long i)                { itsWriter.Int(i);                                                          }
```

Comment:
rather than use the #if it looks cleaner to me to introduce the change as `void saveValue(int32_t i)` . That way on Linux, the implementation with long will automatically use the int64 version as before, and on rtems it uses the int32t_t version.

Reply:
I already tried your suggestion, but unfortunately it does not work on RTEMS. This `#ifdef` should work on any 32-bit system. I hate having `#ifdef`s as well, but I did not find a better solution that works on both 32-bit and 64-bit systems. This fix works on RTEMS, please try if it works on Linux 64-bit as well.